app/components/twilio-make-call.js

```diff
@@ -45,41 +54,65 @@ export default Ember.Component.extend({
     });
 
     twilio_device.disconnect(() => {
-      if(!this.isDestroying && !this.isDestroyed) {
+      if (!this.isDestroying && !this.isDestroyed) {
         this.set("activeCall", false);
         this.get("internetCallStatus").set("activeCall", false);
       }
     });
   },
 
-  actions: {
+  audioDeviceAccessPrompt() {
+    navigator.mediaDevices
+      .getUserMedia({
+        audio: true
+      })
+      .then(function(mediaStream) {
+        this.get("outputSources").set("optional", [
+          { sourceId: mediaStream.id }
+        ]);
+      });
+  },
 
+  actions: {
     makeCall() {
-      var params = { "phone_number": this.get('offerId') + "#" + this.get("currentUserId") };
+      const params = this.get("offerId") + "#" + this.get("currentUserId");
+      const accessToken = this.get("twilioToken");
+      const twilioDevice = this.get("twilio_device");
       this.set("activeCall", true);
       this.get("internetCallStatus").set("activeCall", true);
-      return this.get("twilio_device").connect(params);
+      this.get("isCordovaApp")
+        ? twilioDevice.call(accessToken, params)
+        : twilioDevice.connect({ phone_number: params });
```

Comment:
This is an FYI for when you get the chance or re-visit this code (I'm okay with moving forward with this for now, given time constraints)

A common pattern that is usually used to handle cross-platform compatibility is to create an `"adapter"` or `"wrapper"` which handles internally the device-specific code. Some of the benefits being :
- The code around the app can use a single consistent interface
- Making changes (like an upgrade) can be done in the adapters, without requiring changes all over the app
- Can be exported to other projects